src/universal/modules/welcome/components/WelcomeWizardForms/Step3InviteTeam.js

```diff
@@ -28,9 +28,37 @@ const emailInviteFail = emailsNotDelivered => ({
   level: 'error'
 });
 
-const Step3InviteTeam = props => {
-  const {dispatch, handleSubmit, invitees, inviteesRaw, submitting, teamName, welcome: {teamId}} = props;
-  const onAddInviteesButtonClick = event => {
+
+@reduxForm({
+  form: 'welcomeWizard',
+  destroyOnUnmount: false
+  // TODO: add sync + mailgun async validations
+})
+export default class Step3InviteTeam extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    invitees: PropTypes.array,
+    inviteesRaw: PropTypes.string,
+    onSubmit: PropTypes.func,
+    submitting: PropTypes.bool,
+    teamName: PropTypes.string,
+    welcome: PropTypes.shape({
+      teamId: PropTypes.string,
+    })
+  };
+
+  shouldComponentUpdate(nextProps) {
+    const {teamName, submitting} = nextProps;
+    if (submitting & !teamName) {
+      // do not update if we're submitting
+      return false;
```

Comment:
Sure, I'll walk you through it. `Step[1..3]* ` are all part of the same `WelcomeWizard` form. When the user navigates between pages, we prevent the destruction of form info stored in redux. `teamName` is set in Step 2. It is injected into the container props using `formValueSelector` from `redux-form`:

```javascript
const selector = formValueSelector('welcomeWizard');

const mapStateToProps = state => ({
  /* ... */
  teamName: selector(state, 'teamName'),
  /* ... */
});
```

When we're on Step3 and the user presses the submit button, we dispatch a series of actions: 1) Cashay mutate `inviteTeamMembers`, destroy the form data, push the route to `/team/:teamId`, and finally display a toast.

When the form data is destroyed, teamName becomes null, and then the user will see the string from Step3 flash:

```javascript
<WelcomeHeading copy={<span>Let’s invite some folks to the <b>{teamName}</b> team.</span>}/>
```

If this can be refactored without the thunk this issue goes away. I'll look into it more carefully this morning.